### PR TITLE
 feat: GraphQL program info query and mutation

### PR DESCRIPTION
**Pull request by mayaisrani**
let's try #369 again

### CODE REVIEW

File: oneapp-server/dao/ApplicationProgramInfoDao.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class ApplicationProgramInfoDao extends SQLDataSource {
+  async getProgramInfo(APPLICATION_ID, LANGUAGE) {
+    const con = await this.knex.client.pool.acquire().promise;
+    return this.knex.client.transaction(async (tx) => {
+      const bindVars = {
+        app_id: { dir: oracledb.BIND_IN, val: APPLICATION_ID },
+        lang: { dir: oracledb.BIND_IN, val: LANGUAGE },
+        info: { dir: oracledb.BIND_OUT },
+      };
+      return tx.raw('begin OA_PKG_APP.SP_SELECT_APPLICATION_PROGRAMS(:app_id, :lang, :info); end;', bindVars);
```
Comment (ronaldheft-usds) [inline oneapp-server/dao/ApplicationProgramInfoDao.js:14]: Response should return the first result in the array.

File: oneapp-server/types/ApplicationProgramInfo.js [hunk 2]
```diff
@@ -13,25 +13,25 @@ const typeDef = gql`
...
     "Has active SNAP case"
-    HAVE_ACTIVE_CASE_CURRENTLY: Boolean
+    HAVE_ACTIVE_CASE_CURRENTLY: Boolean!
     "Case number(s) of active SNAP applications"
     CURRENT_CASE_NUMBERS: String
     "Has received SNAP previously"
-    HAD_ACTIVE_CASE_PREVIOULSY: Boolean
+    HAD_ACTIVE_CASE_PREVIOULSY: Boolean!
     "Case number(s) of previous SNAP applications"
     PREVIOUS_CASE_NUMBERS: String
```
Comment (ronaldheft-usds) [inline oneapp-server/types/ApplicationProgramInfo.js:22]: There are additional validation to capture all throughout here. For example, PREVIOUS_CASE_NUMBERS has a regex validation: https://github.com/newjersey/dhs-oneapp-legacy/blob/master/NJOneAppAuto/Forms/Application/Programs.aspx#L483

**Review by ronaldheft-usds — CHANGES_REQUESTED** [review]
Looking good. We should also include validations in these PRs. See my comment here: https://github.com/newjersey/dhs-oneapp/pull/399#pullrequestreview-538660451

File: oneapp-server/dao/ApplicationProgramInfoDao.js (hunk 1 above)
Comment (mayaisrani) [reply oneapp-server/dao/ApplicationProgramInfoDao.js:14]: changed the response value to return the program info object

Comment (ronaldheft-usds) [reply oneapp-server/dao/ApplicationProgramInfoDao.js:14]: Interesting, so specifying a type on the `BIND_OUT` stops the Oracle driver from returning an array?

**Review by ronaldheft-usds — APPROVED** [review]
Assuming the return value isn't returning an array on `getProgramInfo` this looks good.

**Pull request merged**